File: timesheet-calc/python/tests/timesheet_builder.py
from timesheet_calc import Day, Timesheet


class TimesheetBuilder:
    def __init__(self) -> None:
        self._entries: dict[Day, float] = {}

    def with_entry(self, day: Day, hours: float) -> "TimesheetBuilder":
        self._entries[day] = hours
        return self

    def build(self) -> Timesheet:
        return Timesheet(self._entries)

File: timesheet-calc/typescript/tests/timesheetBuilder.ts
import { createTimesheet, Day, type Timesheet } from '../src/timesheet.js';

export class TimesheetBuilder {
  private readonly _entries = new Map<Day, number>();

  withEntry(day: Day, hours: number): this {
    this._entries.set(day, hours);
    return this;
  }

  build(): Timesheet {
    return createTimesheet(this._entries);
  }
}
